addons/l10n_id/models/account_move.py

```diff
@@ -0,0 +1,34 @@
+import requests
+import base64
+from odoo import models
+from odoo.exceptions import UserError
+from odoo.tools.image import image_data_uri
+
+QRIS_API_URL = "https://qris.online/restapi/qris/show_qris.php"
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _generate_qr_code(self, silent_errors=False):
+        """Override
+
+        Using QRIS API in order to generate the QR code
+        """
+        self.ensure_one()
+        if self.company_id.country_code == "ID":
+            params = {
+                "do": "create-invoice",
+                "apikey": self.company_id.l10n_id_qris_apikey,
+                "mID": self.company_id.l10n_id_qris_mid,
+                "cliTrxNumber": self.payment_reference or self.name,
+                "cliTrxAmount": int(self.amount_residual)
+            }
+            res = requests.get(QRIS_API_URL, params=params).json()
```

Comment:
Need to handle the case for timeout issue as well

Reply:
right